genesyscloud/resource_genesyscloud_knowledge_v1_document_test.go

```diff
@@ -0,0 +1,241 @@
+					resource.TestCheckResourceAttr("genesyscloud_knowledge_document."+knowledgeDocumentResource1, "knowledge_document.0.faq.0.answer", answer),
+					resource.TestCheckResourceAttr("genesyscloud_knowledge_document."+knowledgeDocumentResource1, "knowledge_document.0.faq.0.alternatives.0", faqAlternatives2[0]),
+					resource.TestCheckResourceAttr("genesyscloud_knowledge_document."+knowledgeDocumentResource1, "knowledge_document.0.categories.0", knowledgeCategoryName),
+				),
+			},
+			{
+				// Import/Read
+				ResourceName:      "genesyscloud_knowledge_document." + knowledgeDocumentResource1,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+		CheckDestroy: testVerifyKnowledgeV1DocumentDestroyed,
+	})
+}
+
+func generateKnowledgeV1Document(resourceName string, knowledgeCategoryResourceName string, knowledgeBaseResourceName string, languageCode string, varType string, externalUrl string, question string, answer string, faqAlternatives []string, title string, contentLocationUrl string, articleAlternatives []string) string {
+	document := fmt.Sprintf(`
+        resource "genesyscloud_knowledge_v1_document" "%s" {
+            knowledge_base_id = genesyscloud_knowledge_knowledgebase.%s.id
+            language_code = "%s"
+            %s
+        }
+        `, resourceName,
+		knowledgeBaseResourceName,
+		languageCode,
+		generateKnowledgeV1DocumentRequestBodyFaq(varType, externalUrl, question, answer, faqAlternatives, knowledgeCategoryResourceName),
+	)
+	return document
+}
+
+func generateKnowledgeV1DocumentRequestBodyFaq(varType string, externalUrl string, question string, answer string, faqAlternatives []string, knowledgeCategoryResourceName string) string {
+	return fmt.Sprintf(`
+        knowledge_document {
+            type = "%s"
+            external_url = "%s"
+            %s
+            categories = [genesyscloud_knowledge_category.%s.knowledge_category.0.name]
+        }
+        `, varType,
+		externalUrl,
+		generateFaq(question, answer, faqAlternatives),
+		knowledgeCategoryResourceName)
+}
+
+func generateKnowledgeV1DocumentRequestBodyArticle(varType string, externalUrl string, title string, contentLocationUrl string, alternatives []string, knowledgeCategoryResourceName []string) string {
+	return fmt.Sprintf(`
+        knowledge_document {
+            type = "%s"
+            external_url = "%s"
+            %s
+            categories = [genesyscloud_knowledge_category.%s.knowledge_category.0.name]
+        }
+        `, varType,
+		externalUrl,
+		generateArticle(title, contentLocationUrl, alternatives),
+		knowledgeCategoryResourceName)
+}
+
+func generateFaq(question string, answer string, alternatives []string) string {
+	formattedAlternatives := ""
+	for i, alternative := range alternatives {
+		if i > 0 {
+			formattedAlternatives += ", "
+		}
+		formattedAlternatives += strconv.Quote(alternative)
+	}
```

Comment:
It's easier if you pass in an array of quoted strings and say `fmt.Sprintf("x = [%s]", strings.Join(alternatives, ", "))`